fast64_internal/oot/oot_level_writer.py

```diff
@@ -499,7 +518,7 @@ def ootConvertScene(originalSceneObj, transformMatrix, f3dType, isHWv1, sceneNam
     if bpy.context.scene.exportHiddenGeometry:
         hideObjsInList(hiddenObjs)
 
-    roomObjs = [child for child in sceneObj.children if child.data is None and child.ootEmptyType == "Room"]
```

Comment:
How about a check and error if the hierarchy is not correct (room in room, actor directly in scene, etc.)?

Reply:
I don't understand why the diff on github shows ``sceneObj.children``, it should be ``sceneObj.children_recursive``, which checks every objects, so this won't be an issue at all

Reply:
ok I booted my brain, I changed ``children`` to ``children_recursive``, the difference between these two: children don't go through child objects, children_recursive do

so if a room has a weird hierarchy this will check the object, so I don't think it could cause issues

Reply:
Sorry my bad, I think I meant to highlight the line of code underneath this one. My issue is that children_recursive should not be used, because having a room nested in another room doesn't make sense and would cause strange behaviour.

Reply:
I think it'd be better to raise an error if the user tries to export with a room inside a room (and eventually add a warning for that on the UI), because there's the case when you have a room inside an empty with the None type and this wouldn't export that room (unless going through bpy.data but it'd be the same thing in the end)

I understand your concern though, but I can't think of a solution without children_recursive

Reply:
Personally I think rooms being a child of an None type shouldn't be processed either, but if people want to use it that way then I guess its fine.